File: sample/src/main/java/tech/thdev/androidrecyclerview/view/animation/ItemAnimatorEvent.java
package tech.thdev.androidrecyclerview.view.animation;

import android.animation.Animator;

/**
 * Created by Tae-hwan on 20/10/2016.
 */

public class ItemAnimatorEvent {

    public static final int ANIM_START = 0;
    public static final int ANIM_END = 1;

    private final Animator animator;
    private final int event;

    private ItemAnimatorEvent(Animator animator, int event) {
        this.animator = animator;
        this.event = event;
    }

    public static ItemAnimatorEvent eventCreate(Animator animator, int event) {
        return new ItemAnimatorEvent(animator, event);
    }


    public Animator getAnimator() {
        return animator;
    }

    public int getEvent() {
        return event;
    }
}

File: sample/src/main/java/tech/thdev/androidrecyclerview/ui/animation/ItemAnimatorEvent.kt
package tech.thdev.androidrecyclerview.ui.animation

import android.animation.Animator

/**
 * Created by Tae-hwan on 20/10/2016.
 */
class ItemAnimatorEvent private constructor(
    val animator: Animator,
    val event: Int
) {

    companion object {
        const val ANIM_START = 0
        const val ANIM_END = 1

        fun eventCreate(animator: Animator, event: Int): ItemAnimatorEvent {
            return ItemAnimatorEvent(animator, event)
        }
    }

}